File: algorithms/power_of_two.py
class Solution(object):
    def isPowerOfTwo(self, n):
        """
        :type n: int
        :rtype: bool
        """
        return (n != 0) and (abs(n) == (n & (-n)))

t = Solution()

assert(t.isPowerOfTwo(1))
assert(t.isPowerOfTwo(2))
assert(t.isPowerOfTwo(4))
assert(t.isPowerOfTwo(8))
assert(t.isPowerOfTwo(-1024))
assert(not t.isPowerOfTwo(0))
assert(not t.isPowerOfTwo(6))
assert(not t.isPowerOfTwo(7))
assert(not t.isPowerOfTwo(9))
assert(not t.isPowerOfTwo(-10))

print "Tests passed"

File: algorithms/power_of_two.rb
# @param {Integer} n
# @return {Boolean}
def is_power_of_two(n)
  return false if n == 0
  (n & (-n)) == n
end

p is_power_of_two(0)
p is_power_of_two(2)
p is_power_of_two(3)
p is_power_of_two(4)
